Lib/http/server.py

```diff
@@ -1275,7 +1293,19 @@ def test(HandlerClass=BaseHTTPRequestHandler,
                         default=8000, type=int,
                         nargs='?',
                         help='Specify alternate port [default: 8000]')
+    parser.add_argument('--tls-cert',
+                        help='Specify the path to a TLS certificate')
+    parser.add_argument('--tls-key',
+                        help='Specify the path to a TLS key')
     args = parser.parse_args()
+
+    if args.tls_cert is None and args.tls_key is None:
+        tls = None
+    elif not args.tls_cert or not args.tls_key:
+        parser.error('Both --tls-cert and --tls-key must be provided to enable TLS')
+    else:
+        tls = (args.tls_cert, args.tls_key)
```

Comment:
It's common to have the private key and the cert chain in the same file. The key argument should be optional.